doc/userdoc/contribute/nest_with_ides.rst

```diff
@@ -106,3 +106,140 @@ Usage
+
+#. Select a Python interpreter by opening the Command Palette (``Ctrl+Shift+P``) and running
+   **Python: Select Interpreter**. Select the Python installation found by NEST in the configuration step.
+#. Open ``File > Preferences > Settings``, go to **Terminal>integrated>Env:<your OS>**, and click on
+   **Edit in settings.json**. VS Code will open ``settings.json`` and create a JSON object (JSON objects are
+   similar to dictionaries in Python). In that object, add
+
+   .. code-block:: JSON
+
+      "PYTHONPATH": "${workspaceFolder}/../build/install/lib/<YOUR PYTHON VERSION>/site-packages:${env:PYTHONPATH}"
+
+   replacing ``<YOUR PYTHON VERSION>`` with your Python version, e.g. ``python3.8``. You can check the
+   ``build/install/lib/`` directory to get the correct name.
+#. Open or create a Python script.
+#. When you open a Python file for the first time, VS Code will ask if you want to install a linter. It is
+   recommended to install a linter, for example *Flake8*, to keep the code clean and readable.
+#. Run the script by clicking the triangle (▷) in the top right corner, or right-clicking in the editor and choosing
+   **Run Python file in Terminal**. A panel should open with a terminal showing the output.
+
+Running a NEST Python script with a Python debugger
+###################################################
+
+The steps below give a rough guide to how you can run a NEST Python script with the built-in debugger. For more detailed
+documentation on Python debugging in VS Code, see the
+`VS Code Python debugging documentation <https://code.visualstudio.com/docs/python/debugging>`_.
+
+#. Set up the interpreter and script as described above.
+#. In the Side Bar, open the **Run** pane, or press ``Ctrl+Shift+D``.
+#. Create a Python debug config by either
+
+   * selecting **Add configuration...** from the dropdown menu, or
+   * clicking the ``Create a launch.json file`` link, if the ``launch.json`` doesn't exist
+#. The debug configuration defaults to the current open Python file. Go back to the Python script and start the
+   debugger by selecting the debug configuration from the dropdown in the Run pane (you can also use
+   the hotkey ``F5``).
+#. A panel with output will open, and the program will run until it finishes, or encounters an error or a breakpoint.
+
+Running a SLI script with a debugger
+####################################
+
+The steps below give a rough guide to how you can run NEST with GDB in VS Code. For more detailed
+documentation on C++ debugging in VS Code, see the
+`VS Code C++ debugging documentation <https://code.visualstudio.com/docs/cpp/cpp-debug>`_.
+
+#. In the Side Bar, open the **Run** pane, or press ``Ctrl+Shift+D``.
+#. Add a debug config by either
+
+   * selecting **Add configuration...** from the dropdown menu, or
+   * clicking the ``Create a launch.json file`` link, if the ``launch.json`` doesn't exist
+#. Choose the template for ``C/C++ (gdb) launch`` and
```

Comment:
In the case of Mac, the default C++ debugger is `lldb` and the template to choose is `C/C++ (lldb) launch`. It might be useful to add this here?

Reply:
Added.